extension/data.ts

```diff
@@ -278,7 +278,7 @@ class RcxDict {
             if (type & rule.typeMask && end === rule.from) {
               const newWord =
                 word.substr(0, word.length - rule.from.length) + rule.to;
-              if (newWord.length <= 1) {
+              if (newWord.length <= 0) {
```

Comment:
Copying from my non review comment so it's easier to notice:

> We should also add some tests for the data.ts changes. There aren't tests for that file yet but it should be fairly straightforward to do in the same manner of other files like background.ts.
> 
> That being said, it can be tricky writing the first tests for a class, so let me know if you want me to add a commit with some tests this time around.

Reply:
Yes, I would like some help for a commit with some tests as I am not entirely sure on how to do that, thanks!

Reply:
I added some tests for the new branches as per the latest commit. The first tests for a method are always a bit tricky so I had to install some new assertion helpers to write tests to the level of readability and robustness that I wanted.

Please read them for your own reference; will definitely be useful if you make other changes in the future! (And also useful as a general software development example.)